src/ui/pages/GetStarted/ImportWallet/MnemonicImportView/getFirstNMnemonicWallets.worker.ts

```diff
@@ -1,50 +1,97 @@
 import { HDNode } from '@ethersproject/hdnode';
-import type { MaskedBareWallet } from 'src/shared/types/BareWallet';
-import { getAccountPath } from 'src/shared/wallet/derivation-paths';
+import { Keypair } from '@solana/web3.js';
+import * as bip39 from 'bip39';
+import { derivePath } from 'ed25519-hd-key';
+import type { BareWallet, MaskedBareWallet } from 'src/shared/types/BareWallet';
+import type { DerivationPathType } from 'src/shared/wallet/derivation-paths';
+import {
+  getAccountPathEthereum,
+  getAccountPath,
+} from 'src/shared/wallet/derivation-paths';
 import type { LocallyEncoded } from 'src/shared/wallet/encode-locally';
 import {
   decodeMasked,
   encodeForMasking,
 } from 'src/shared/wallet/encode-locally';
+import { fromSolanaKeypair } from 'src/modules/solana/keypairs';
 
 export interface Params {
+  /**
+   * 'ecdsa' is used to create ethereum key pairs;
+   * 'ed25519' is used to create solana key pairs;
+   */
+  curve: 'ecdsa' | 'ed25519';
```

Comment:
Do we need to set up curve as param? May be it is gonna be easier to set ecosystem here and use related curve in the very last step?

Reply:
Hard to make a correct guess at this point. I think `curve` is a fair name since this is closer to what's actually different in the algorithms. I think with more blockchains the terms can be refined